storage/innobase/handler/ha_innodb.cc

```diff
@@ -3830,12 +3840,14 @@ innobase_init(
 
 	/* The buffer pool needs to be able to accommodate enough many
 	pages, even for larger pages */
-	if (UNIV_PAGE_SIZE > UNIV_PAGE_SIZE_DEF
-	    && innobase_buffer_pool_size < (24 * 1024 * 1024)) {
+	srv_buf_pool_min_size= min_buffer_pool_size();
```

Comment:
This assignment does not appear to be reflected by `mysql-test/suite/sys_vars/r/sysvars_innodb.result`, which is (unsurprisingly) showing the changed initial value of 2 megabytes when using the default page size of 16 kilobytes. I think that this assignment should better refer to `MYSQL_SYSVAR(buffer_pool_size).min_val` if that is possible. Ideally, the reported minimum value for the default `innodb_page_size=16k` would not change.

Reply:
> This assignment does not appear to be reflected by `mysql-test/suite/sys_vars/r/sysvars_innodb.result`, which is (unsurprisingly) showing the changed initial value of 2 megabytes when using the default page size of 16 kilobytes. I think that this assignment should better refer to `MYSQL_SYSVAR(buffer_pool_size).min_val` if that is possible.

Doing that now. I don't think its complete.

> Ideally, the reported minimum value for the default `innodb_page_size=16k` would not change.
I don't think this is possible (unless we make 5M the absolute minimum on all page sizes). The command line args are passed in order and the resize happens before innodb code even starts.
```
$  sql/mysqld --innodb-buffer-pool-chunk-size=1M --innodb-buffer-pool-size=1M --innodb-page-size=4k  --datadir=/tmp/${PWD##*/}-datadir4
2022-01-18 19:45:37 140087854176192 [Note] sql/mysqld (mysqld 10.2.42-MariaDB) starting as process 202208 ...
2022-01-18 19:45:37 140087854176192 [Warning] option 'innodb-buffer-pool-size': signed value 1048576 adjusted to 5242880
2022-01-18 19:45:37 140087854176192 [Note] InnoDB: innodb_page_size=4096
```